### PR TITLE
feat: create base modal component

### PR DESCRIPTION
**Pull request by Ibwedagama**
### Overview
as described by the title

**Usage**

```vue
<template>
  <div>
    <button @click="openModal">Open Modal</button>

    <!-- Base Modal -->
    <BaseModal
      :open="modal.isOpen"
      :with-close-button="true"
      @close="closeModal"
    >
      <ModalTitle>
        Lorem ipsum dolor sit amet consectetur adipisicing elit.
      </ModalTitle>
      <ModalBody>
        <p>
          Lorem ipsum dolor, sit amet consectetur adipisicing elit. Rem facere
          aliquid iusto. Eligendi minus adipisci eos consectetur officia nostrum
          vero iste? Minus reiciendis, at numquam alias, tenetur nulla suscipit
          qui beatae voluptas sint odit, quidem autem non. Est veritatis, enim
          adipisci facere odit ad id vitae, nostrum voluptatibus molestiae
          numquam.
        </p>
      </ModalBody>
      <ModalFooter>
        <button
          class="w-full rounded-lg bg-green-800 py-2 text-white"
          @click="closeModal"
        >
          Close
        </button>
      </ModalFooter>
    </BaseModal>
  </div>
</template>

<script setup lang="ts">
  const modal = reactive({ isOpen: false })

  const closeModal = () => {
    modal.isOpen = false
  }

  const openModal = () => {
    modal.isOpen = true
  }
</script>
```

### Changes
- create `BaseModal` component
- create `ModalTitle` component
- create `ModalBody` component
- create `ModalFooter` component
- add `close.svg` icon

### Preview

https://github.com/jabardigitalservice/j-site-builder/assets/33661143/bd455520-ff0e-4671-9b8e-799e8fb26359

### Evidence

title: feat create base modal component
project: J-Site
participants: @Ibwedagama @yoslie @maulanayuseph @agunghide

### CODE REVIEW

**Review by agunghide — APPROVED** [review]

File: components/ModalFooter.vue [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+      'justify-start': props.position === 'left',
+      'justify-end': props.position === 'right',
+    }"
+  >
+    <slot />
+  </div>
+</template>
+
+<script setup lang="ts">
+  type PositionTypes = 'center' | 'left' | 'right'
```
Comment (yoslie) [inline components/ModalFooter.vue:15]: can u explained what is this `PositionTypes` and it's use   🤔

**Review by yoslie — APPROVED** [review]

File: components/ModalFooter.vue (hunk 1 above)
Comment (Ibwedagama) [reply components/ModalFooter.vue:15]: yes @yoslie , so the `position`  props have a type of String, and it has predefined values 'center, left, and right'

so using Typescript we can add type to the props like this

```js
 position: {
      type: String as () => PositionTypes,
      default: 'center',
    },
```

so when user using the component and use `position` prop, the autocomplete will show what value available for this props, and when user using value that is not available Typescript will throw an error

Comment (yoslie) [reply components/ModalFooter.vue:15]: I mean why dont use `const` instead? 🤣

Comment (Ibwedagama) [reply components/ModalFooter.vue:15]: `const` is a variable, and type is a type 😆 @yoslie 

it is something that is used by Typescript to know the types of variable or object properties

Comment (Ibwedagama) [reply components/ModalFooter.vue:15]: this is the power of Typescript 😆

 

https://github.com/jabardigitalservice/j-site-builder/assets/33661143/921b25cd-b37b-4874-9545-f32e50b035e9

@yoslie

**Review by maulanayuseph — APPROVED** [review]

**Pull request merged**